pkg/getty/usr/bin/rungetty.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+set -x
+
+infinite_loop() {
+	while true; do
+		$@
+	done
+}
+
+# run getty on all known consoles - except those already in inittab
+start_getty() {
+	tty=${1%,*}
+	speed=${1#*,}
+	inittab="$2"
+	securetty="$3"
+	line=
+	term="linux"
+	[ "$speed" = "$1" ] && speed=115200
+
+	case "$tty" in
+	ttyS*|ttyAMA*|ttyUSB*|ttyMFD*)
+		line="-L"
+		term="vt100"
+		;;
+	tty?)
+		line=""
+		speed="38400"
+		term=""
+		;;
+	esac
+
+	# are we secure or insecure?
+	loginargs=
+	if [ "$INSECURE" == "true" ]; then
+		loginargs="-n -l /bin/sh"
+	fi
+
+	# skip consoles already in inittab
+	#grep -q "^$tty:" "$inittab" && return
+
+	#echo "$tty::respawn:/sbin/getty -n -l /bin/sh $line $speed $tty $term" >> "$inittab"
+	if ! grep -q -w "$tty" "$securetty"; then
+		echo "$tty" >> "$securetty"
+	fi
+	# respawn forever
+	infinite_loop setsid -w /sbin/getty $loginargs $line $speed $tty $term &
+}
+
+# check if we have /etc/getty.shadow
+ROOTSHADOW=/proc/1/root/etc/getty.shadow
```

Comment:
> If you can run ctr you can bind mount /etc/getty.shadow into a container and inspect it.

If you can run `ctr` *and* have access to the host namespace *and* are root. Basically anyone with `pid:host` can see it (which is true anyways via `/proc/1/root/etc/getty.shadow` as you pointed out).

So let's leave the security of `/etc/shadow` out of it, and focus on the user flow.

Never really liked `--env-file` type things much if they couldn't be recorded into the "code" (i.e. yml)....

Is there no saner way to bind-mount this? What if we bind-mounted `/etc:/hostetc`, and then we don't need to bind-mount, just see if the file exists? `cp /hostetc/getty.shadow /etc/shadow`?